Sources/StreamChat/Utils/Codable+Extensions.swift

```diff
@@ -4,12 +4,26 @@
 
 import Foundation
 
+final class StreamJSONDecoder: JSONDecoder {
+    override func decode<T>(_ type: T.Type, from data: Data) throws -> T where T: Decodable {
```

Comment:
What about having another name for it? And then you have it in a fileprivate extension in the UserPayload?

Reply:
I don't see a need for a subclass here 🤔

Reply:
I don't follow? The aim is to make sure cache is reset before any decoding session, since cache is only valid for a single decoding session

Reply:
If you have a fileprivate extension (in UserPayload), you avoid overriding this method, and only add this logic there.
That avoids having a subclass, too.

Reply:
fileprivate extension to which class exactly? And where would we call this new func? Where would we reset the cache?

Reply:
Forget about it, I was misled by the same signature in JSONDecoder and its container.

Worth adding a comment there on the overriden `decode` function to call out it is only affecting the top level payload/data, as the children of the payload call `decode` on the container, and not on the decoder.

Reply:
This cache invalidation doesn't seem right 🤔 Does this mean that for a session, if any user updates their data, you won't get the data correctly? 

Example:
1. You fetch the message list with user avatars
2. User updates the avatar, this updates local DB correclty
3. you fetch again the message list, since the user is in the cache, it overrides the updated user
4. You get the data incorrectly

It seems to me that instead of removing all cache here, can we remove the user payload from the cache whenever we have a user updated event?

Reply:
@nuno-vieira Cache is invalidated before and after any decoding session, so it's not possible for it to become invalid. I've added thread safety checks for this too, to make sure cache doesn't leak inter-session. You can QA this branch to see that the scenario you're describing doesn't happen.